### PR TITLE
Fix extra blank line in sequence

### PR DESCRIPTION
**Pull request by Julow**
Fix https://github.com/ocaml-ppx/ocamlformat/issues/1019

### CODE REVIEW

File: src/Fmt_ast.ml [hunk 1]
```diff
@@ -541,6 +541,23 @@ let fmt_assign_arrow c =
...
+  | `Preserve_one ->
+      let l1 = xe1.ast.pexp_loc.loc_end.pos_lnum in
+      let rec blank_lines l2 =
+        let open Location in
+        let n = l2.loc_start.pos_lnum - l1 in
+        if n <= 0 then 0
+        else if Cmts.has_before c.cmts l2 then
+          let s = l2.loc_start in
+          let loc_start = {s with pos_lnum= s.pos_lnum - 1} in
+          blank_lines {l2 with loc_start}
```
Comment (Julow) [inline src/Fmt_ast.ml]: I'm not a fan of constructing a record from OCaml's parser but it's the only way to talk to `Cmts`.

Comment (gpetiot) [reply src/Fmt_ast.ml]: Could we instead use the location `Cmt.loc`? (not sure iterating on the comments from one location is much cleaner though)

Comment (Julow) [reply src/Fmt_ast.ml]: I completely overlooked that. I have a much cleaner implementation now that only compare line numbers and is more readable.

File: src/Cmts.mli [hunk 2]
```diff
@@ -130,6 +130,9 @@ val remaining_comments : t -> (Cmt.t * string * Sexp.t) list
 
 val remaining_locs : t -> Location.t list
 
+val remaining_before : t -> Location.t -> Cmt.t list
+(** Returns comments before [loc] *)
```
Comment (gpetiot) [inline src/Cmts.mli]: I would maybe change to something like: `[remaining_before c loc] returns comments before [loc].`

Comment (Julow) [reply src/Cmts.mli]: Yes that's better.

**Pull request merged**